lib/Generator.php

```diff
@@ -0,0 +1,99 @@
+<?php
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2018, Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * @author Roeland Jago Douma <roeland@famdouma.nl>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Password_Policy;
+
+use OC\HintException;
+use OCP\Security\ISecureRandom;
+
+class Generator {
+
+	/** @var PasswordPolicyConfig */
+	private $config;
+
+	/** @var PasswordValidator */
+	private $validator;
+
+	/** @var ISecureRandom */
+	private $random;
+
+	public function __construct(PasswordPolicyConfig $config,
+								PasswordValidator $validator,
+								ISecureRandom $random) {
+		$this->config = $config;
+		$this->validator = $validator;
+		$this->random = $random;
+	}
+
+	public function generate(): string {
+
+		$lenght = $this->config->getMinLength();
+
+		$password = '';
+		$chars = '';
+
+		while (true) {
```

Comment:
I would love to see a max iteration here to not run into CPU burning stuff if there is a special rule set up that can't be accomplished by this code. Something like 20 iterations.

Reply:
Yes makes sense. I really doubt it will happen (as you can see the password just gets longer and longer) but I guess it could in some cases.